share/smack/include/pthread.h

```diff
@@ -0,0 +1,71 @@
+//
```

Comment:
Ok, I think I finally see what the issue is: that other system headers (i.e., other than `pthread.h`) may provide conflicting type definitions for pthread API types, e.g., in `#include <sys/types.h>`. (I guess that is what was originally meant by “recursively [including] pthread typedefs.”)

But could there be a portable way to handle this? I doubt the current solution of hijacking `pthreadtypes.h` will work across platforms, since `pthreadtypes.h` is a very platform dependent entity. I could be wrong though.

So it seems that fixing pthread API types to our liking in a robust manner is hard. Hijacking system definitions does not seem portable, nor universally feasible. For example the `<sys/types.h>` header on OS X defines `typedef __darwin_pthread_t pthread_t` guarded by the `_PTHREAD_T` and `__POSIX_LIB__` macros. Perhaps defining the `_PTHREAD_T` and `__POSIX_LIB__` macro would prevent this definition, but would it prevent all possible definitions of `pthread_t` in all possible OS X system headers? I wonder if there could be a set of macros, e.g., including `_PTHREAD_T` for OS X and `PTHREADTYPES_H` on Linux, that would ensure that no system header defines `pthread_t`?